test/mixer/server.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mixer
+
+import (
+	"fmt"
+	"sync/atomic"
+
+	"github.com/golang/protobuf/ptypes"
+	google_rpc "github.com/googleapis/googleapis/google/rpc"
+	"golang.org/x/net/context"
+
+	"istio.io/pilot/test/mixer/pb"
+)
+
+// Server is a basic Mixer server
+type Server struct {
+	words   []string
+	counter uint64
+}
+
+// Check implements service check
+func (s *Server) Check(ctx context.Context, msg *istio_mixer_v1.CheckRequest) (*istio_mixer_v1.CheckResponse, error) {
+	atomic.AddUint64(&s.counter, 1)
+	bag := s.decode(msg.Attributes)
+	fmt.Printf("[%d] words.size=%d\n", s.counter, len(msg.Attributes.Words))
+	for k, v := range bag {
+		fmt.Printf("[%d] %s=%s\n", s.counter, k, v)
+	}
+	return &istio_mixer_v1.CheckResponse{
+		Precondition: &istio_mixer_v1.CheckResponse_PreconditionResult{
+			Status: &google_rpc.Status{Code: 0},
+		},
+	}, nil
```

Comment:
Is it possible to mock this or add a parameter to this such that we can decide when Mixer should fail or accept a check request? Would come in handy during testing stability

Reply:
Yeah, we can do absolutely anything now that we can have a server endpoint. This is just a start to get data path validation, before enforcing control behavior. I'm thinking of adding control HTTP endpoints on this to mark start/end of a test, and then send rules for asserting check response.